include/hermes/async_engine.hpp

```diff
@@ -215,11 +215,20 @@ class async_engine {
 
         // if address contains a prefix, make sure that it matches 
         // the transport protocol used by the engine
-
-        const std::size_t pos = addr.find("://");
+        const auto pos = addr.rfind("://");
 
         if(pos != std::string::npos) {
-            if(addr.substr(0, pos+3) != get_transport_lookup_prefix(m_transport)) {
+            // multiple URIs may be part of address string (e.g., if auto_sm is used)
+            auto pos_delim = addr.rfind(';');
```

Comment:
Mercury devs changed the delimiter to `#` in commit mercury-hpc/mercury@a4bc1972c4c9ed98b596e49b4c41a9c2bac15598. Since they don't provide a public define for the delimiter (`HG_CORE_ADDR_DELIMITER` is defined privately in `mercury_core.c`), we should probably either check for the Mercury version and choose the appropriate delimiter, or try with one and if it's not found try with the other one.

Reply:
[`std::find_first_of()`](https://en.cppreference.com/w/cpp/algorithm/find_first_of) might be of help here.

Reply:
Solved with: 37200369feed9b04d5a6dec8a51756fbc5908a5e

Uses two rfind()s (backwards search), starting with newer Mercury version first.